pkg/client/httputils.go

```diff
@@ -165,7 +165,25 @@ func (rb *HTTPRequestBuilder) JSONResDoWithRetries(ret any, retryOpts RetryOptio
 		return err
 	}
 	defer res.Body.Close()
-	return rb.parseResponse(res, ret)
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+	decoder := json.NewDecoder(bytes.NewReader(b))
+	if res.StatusCode >= 200 && res.StatusCode <= 299 {
+		if ret == nil {
+			return nil
+		}
+		if err := decoder.Decode(ret); err != nil {
+			return fmt.Errorf("failed decoding successful response(%d), body: %s, error: %w", res.StatusCode, string(b), err)
+		}
+		return nil
```

Comment:
Is it important that a successful response decodes to valid json when the response is dropped anyway?

Reply:
My mistake, I didn't realize the caller of this function similarly has access to `ret`.

Reply:
no problem! thanks for reviewing.